### PR TITLE
fix: don't crash cli on older versions of node

### PR DESCRIPTION
**Pull request by threepointone**
The entry point for the cli `./bin/wrangler.js` wasn't being transpiled, which meant that any newer syntax would cause it to crash on older versions of node. This is a problem because we need this script to run even on unsupported versions so we can log a message telling the user their version of node is too old. The fix here is to just transpile the entry point as well.

Additionally, this fix also runs `emit-types` only when doing a full build, so local dev is a little faster.

Fixes https://github.com/cloudflare/wrangler2/issues/1443
Fixes https://github.com/cloudflare/wrangler2/issues/1435

### CODE REVIEW

File: packages/wrangler/scripts/bundle.ts [hunk 1]
```diff
@@ -54,19 +54,35 @@ async function buildMiniflareCLI(flags: BuildFlags = {}) {
 	});
 }
 
+async function buildCLIEntryPoint(flags: BuildFlags = {}) {
+	await build({
+		entryPoints: ["./bin-src/wrangler.js"],
+		outfile: "./bin/wrangler.js",
+		platform: "node",
+		format: "cjs",
+		target: "node12",
```
Comment (rozenmd) [inline packages/wrangler/scripts/bundle.ts:63]: I assume we're picking node12 since it's the most recent one to go EOL?

Comment (threepointone) [reply packages/wrangler/scripts/bundle.ts:63]: just because it was the one referenced in the original issue, but I could go earlier too. lemme choose node10.

File: .changeset/cuddly-turkeys-love.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+"wrangler": patch
+---
+
+fix: don't crash cli on older versions of node
+
+The entry point for the cli `./bin/wrangler.js` wasn't being transpiled, which meant that any newer syntax would cause it to crash on older versions of node. This is a problem because we need this script to run even on unsupported versions so we can log a message telling the user their version of node is too old. The fix here is to just transpile the entry point as well.
+
+Additionally, this fix also runs `emit-types` only when doing a full build, so local dev is a little faster.
```
Comment (petebacondarwin) [inline .changeset/cuddly-turkeys-love.md:9]: Feels like this should be two changesets since the two changes are completely unrelated?

File: packages/wrangler/bin-src/wrangler.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 #!/usr/bin/env node
-const { spawn } = require("child_process");
-const path = require("path");
-const fs = require("fs");
-const os = require("os");
-const semiver = require("semiver");
+import { spawn } from "child_process";
```
[the remark this answers is not in the document]
Comment (threepointone) [reply packages/wrangler/bin-src/wrangler.js:2]: well maybe paths will be fine, lemme see

File: .changeset/cuddly-turkeys-love.md (hunk 2 above)
Comment (threepointone) [reply .changeset/cuddly-turkeys-love.md]: done

File: packages/wrangler/bin-src/wrangler.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (threepointone) [reply packages/wrangler/bin-src/wrangler.js]: ok, both done

**Review by petebacondarwin — APPROVED** [review]
LGTM!

**Comment by threepointone** [comment]
Hmm, the test failure is weird, looks like `wrangler` isn't available in some fixtures. Investigating.

**Comment by threepointone** [comment]
Closing this in favour of https://github.com/cloudflare/wrangler2/pull/1463
